src/app/shared/abstracts/base-product/base-product.component.ts

```diff
@@ -1,62 +1,35 @@
-import { Component, Input } from "@angular/core";
+import { Component, Input, OnInit } from "@angular/core";
 import { FormBuilder, Validators } from "@angular/forms";
 
+import { CartModifier } from "@shared/models/types/cart-modifier.interface";
 import { Product } from "@shared/models/classes/product.class";
-import { Roast } from "@shared/models/types/roast.type";
-import { Weight } from "@shared/models/types/weight.type";
-import { CustomOption } from "@shared/models/types/custom-option.interface";
-
-import { ROAST_OPTIONS, WEIGHT_OPTIONS } from "@utils/constants/form-options";
 
 @Component({ template: "" })
-export abstract class BaseProductComponent {
+export abstract class BaseProductComponent implements OnInit {
     @Input() product!: Product;
-
-    private readonly _weightOptions = WEIGHT_OPTIONS;
-    readonly defaultWeight: Weight = this._weightOptions[0];
-
-    private readonly _roastOptions = ROAST_OPTIONS;
-    readonly defaultRoast: Roast = this._roastOptions[0];
-
-    optionsForm = this.fb.group({
-        weight: [this.defaultWeight, Validators.required],
-        roast: [this.defaultRoast, Validators.required],
-        quantity: [1, Validators.required],
-    });
-
-    get weight(): string {
-        return this.optionsForm.get("weight")!.value;
-    }
-
-    get roast(): string {
-        return this.optionsForm.get("roast")!.value;
-    }
+    optionsForm = this.fb.group({});
 
     get quantity(): number {
         return this.optionsForm.get("quantity")!.value;
     }
 
-    get roastOptions(): string[] {
-        return this._roastOptions.map(r => r as string);
-    }
-
-    get weightOptions(): string[] {
-        return this._weightOptions.map(w => w as string);
+    get productAttributes() {
+        return this.product
+            .attributes
+            .map(attr => attr.name);
     }
 
-    get snipcartOptions(): CustomOption[] {
-        return [
-            {
-                name: "Weight",
-                list: this.weightOptions,
-                selection: this.weight,
-            },
-            {
-                name: "Roast",
-                list: this.roastOptions,
-                selection: this.roast,
-            },
-        ];
+    get cartModifiers(): CartModifier[] {
+        return this.product
+            .attributes
+            .map(
+                attr => {
+                    return {
```

Comment:
you can use an implicit return also with objects. Simply use a leading (

.map(attr => ({})